src/sqlite3/result_set.cr

```diff
@@ -44,25 +44,57 @@ class SQLite3::ResultSet < DB::ResultSet
         raise Exception.new(sqlite3_statement.sqlite3_connection)
       end
     @column_index += 1
-    value
+    value.as(Float64 | Int64 | Slice(UInt8) | String | Time | Nil)
```

Comment:
Because that how it might be used in ORMs. And I think it's logical to use `DB::ResultSet` instead of `SQLite3::ResultSet`.

Example of usage https://github.com/fridgerator/crecto/blob/master/src/crecto/repo.cr#L341

Also, the issue and solution is described here https://github.com/crystal-lang/crystal/issues/4176